engine/testabilities/fixture_engine.go

```diff
@@ -159,28 +170,77 @@ func (f *engineFixture) Tx() txtestability.TransactionSpec {
 
 // prepareDBConfigForTests creates a new connection that will be used as connection for engine
 func (f *engineFixture) prepareDBConfigForTests() {
-	require.Equal(f.t, datastore.SQLite, f.config.Db.Datastore.Engine, "Other datastore engines are not supported in tests (yet)")
-
-	// It is a workaround for development purpose to check the code with postgres instance.
-	if ok, dbName := testmode.CheckPostgresMode(); ok {
-		f.config.Db.Datastore.Engine = datastore.PostgreSQL
-		f.config.Db.SQL.User = "postgres"
-		f.config.Db.SQL.Password = "postgres"
-		f.config.Db.SQL.Name = dbName
-		f.config.Db.SQL.Host = "localhost"
+	if f.tryDevelopmentPostgres() {
 		return
 	}
 
-	// It is a workaround for development purpose to check what is the db state after running a tests.
-	if testmode.CheckFileSQLiteMode() {
-		f.dbConnectionString = fileDbConnectionString
+	if f.tryDevelopmentSQLite() {
+		return
+	}
+
+	// Default: use SQLite in-memory
+	f.useSQLite()
+}
+
+// tryDevelopmentPostgres configures a development PostgreSQL connection if requested
+// Returns true if this configuration was applied
+func (f *engineFixture) tryDevelopmentPostgres() bool {
+	ok, dbName := testmode.CheckPostgresMode()
+	if !ok {
+		return false
+	}
+
+	f.config.Db.Datastore.Engine = datastore.PostgreSQL
+	f.config.Db.SQL.User = "postgres"
+	f.config.Db.SQL.Password = "postgres"
+	f.config.Db.SQL.Name = dbName
+
+	host, port := os.Getenv(testmode.EnvDBHost), os.Getenv(testmode.EnvDBPort)
+	if host != "" && port != "" {
+		f.config.Db.SQL.Host = host
+		f.config.Db.SQL.Port = port
```

Comment:
what if someone just want to set the host, and keep default port, and vice versa?
As you're already adding this feature, it would be good to make it more flexible